x-pack/filebeat/input/httpjson/request.go

```diff
@@ -146,38 +168,170 @@ func (rf *requestFactory) newHTTPRequest(stdCtx context.Context, trCtx *transfor
 type requester struct {
 	log               *logp.Logger
 	client            *httpClient
-	requestFactory    *requestFactory
+	requestFactories  []*requestFactory
 	responseProcessor *responseProcessor
 }
 
 func newRequester(
 	client *httpClient,
-	requestFactory *requestFactory,
+	requestFactory []*requestFactory,
 	responseProcessor *responseProcessor,
 	log *logp.Logger) *requester {
 	return &requester{
 		log:               log,
 		client:            client,
-		requestFactory:    requestFactory,
+		requestFactories:  requestFactory,
 		responseProcessor: responseProcessor,
 	}
 }
 
-func (r *requester) doRequest(stdCtx context.Context, trCtx *transformContext, publisher inputcursor.Publisher) error {
-	req, err := r.requestFactory.newHTTPRequest(stdCtx, trCtx)
+// collectResponse returns response from provided request
+func (rf *requestFactory) collectResponse(stdCtx context.Context, trCtx *transformContext, r *requester, publisher inputcursor.Publisher) (*http.Response, error) {
+	req, err := rf.newHTTPRequest(stdCtx, trCtx)
 	if err != nil {
-		return fmt.Errorf("failed to create http request: %w", err)
+		return nil, fmt.Errorf("failed to create http request: %w", err)
 	}
-
 	httpResp, err := r.client.do(stdCtx, trCtx, req)
 	if err != nil {
-		return fmt.Errorf("failed to execute http client.Do: %w", err)
+		return nil, fmt.Errorf("failed to execute http client.Do: %w", err)
+	}
+	return httpResp, nil
+}
+
+// generateNewUrl returns new url value using replacement from oldUrl with ids
+func generateNewUrl(replacement *regexp.Regexp, oldUrl, ids string) (url.URL, error) {
+	newUrl, err := url.Parse(replacement.ReplaceAllString(oldUrl, ids))
+	if err != nil {
+		return url.URL{}, fmt.Errorf("failed to replace value in url: %w", err)
+	}
+	return *newUrl, nil
+
+}
+
+func (r *requester) doRequest(stdCtx context.Context, trCtx *transformContext, publisher inputcursor.Publisher) error {
+	var (
+		n                 int
+		ids               []string
+		err               error
+		split             *split
+		urlCopy           url.URL
+		urlString         string
+		httpResp          *http.Response
+		intermediateResps []*http.Response
+		finalResps        []*http.Response
+	)
+	for i, rf := range r.requestFactories {
+		// iterate over collected ids from last response
+		if i == 0 {
+			// perform and store regular call responses
+			httpResp, err = rf.collectResponse(stdCtx, trCtx, r, publisher)
+			if err != nil {
+				return fmt.Errorf("failed to execute rf.collectResponse: %w", err)
+			}
+			if len(r.requestFactories) == 1 {
+				finalResps = append(finalResps, httpResp)
+				n, err = r.processAndPublishEvents(stdCtx, trCtx, publisher, finalResps, true)
+				if err != nil {
+					return err
+				}
+				continue
+			} else {
```

Comment:
The else block is not required since in the `len(r.requestFactories) == 1` case you hit the `continue`, so remove it and de-indent the code in the body.

Reply:
Right